File: 2021/day02/solve.c
#include "stdio.h"
#include "string.h"

int main() {
    FILE *file = fopen("input.txt", "r");
    int x,y = 0;
    char line[32] = "";
    while(fgets(line, sizeof(line), file)) {
        char command[32] = "";
        int movement;
        sscanf(line, "%s %d", command, &movement);
        if (strcmp(command, "forward") == 0) {
            x += movement;
        } else if (strcmp(command, "up") == 0) {
            y -= movement;
        } else if (strcmp(command, "down") == 0) {
            y += movement;
        } else {
            return 1;
        }
    }
    printf("Sumbarine position X:%d Y:%d Mul: %d", x, y, x * y);
    fclose(file);
    return 0;
}

File: 2022/day18/solve.py
data = set(line.strip() for line in open("input.txt"))
p = 0

mask = [
    (1, 0, 0),
    (-1, 0, 0),
    (0, 1, 0),
    (0, -1, 0),
    (0, 0, 1),
    (0, 0, -1)
]

for x, y, z in data:
    p += 6
    for dx, dy, dz in mask:
        if (x+dx, y+dy, z+dz) in data:
            p -= 1

print(p)
